### PR TITLE
stitch dependencies to assets post build

### PR DESCRIPTION
**Pull request by Nikola-3**
Some functionality available to plugins depends on having dependencies directly available on each asset. This PR intends to make them available, after the rust build to avoid duplicating info for serialisation.

### CODE REVIEW

File: packages/core/core/src/requests/AssetGraphRequestRust.js [hunk 1]
```diff
@@ -86,6 +87,45 @@ export function createAssetGraphRequestRust(
...
+    ContentKey,
+    | {type: 'dependency', value: Set<Dependency>}
+    | {type: 'asset', value: Asset},
+  >,
+) {
+  const seenResult = seenAssetMap.get(id);
+
+  if (entity.type === 'asset') {
+    const asset = entity.value;
+    asset.dependencies = asset.dependencies ?? new Map();
```
Comment (MonicaOlejniczak) [inline packages/core/core/src/requests/AssetGraphRequestRust.js:105]: Just set this with the other setters, after `asset.contentKey = id;`

File: packages/core/core/src/requests/AssetGraphRequestRust.js [hunk 2]
```diff
@@ -86,6 +87,45 @@ export function createAssetGraphRequestRust(
   });
 }
 
+export function trackDependenciesPerAsset(
```
Comment (MonicaOlejniczak) [inline packages/core/core/src/requests/AssetGraphRequestRust.js:90]: I find this code pretty convoluted, might help to split up the logic with separate maps as well

File: packages/core/core/src/requests/AssetGraphRequestRust.js [hunk 3]
```diff
@@ -86,6 +87,45 @@ export function createAssetGraphRequestRust(
...
+        seenResult?.type === 'dependency' ? seenResult.value : new Set();
+      existingDeps.add(dependency);
+      seenAssetMap.set(id, {type: 'dependency', value: existingDeps});
+    } else {
+      seenResult.value.dependencies.set(dependency.id, dependency);
+    }
+  }
+}
+
 function getAssetGraph(serializedGraph, options) {
```
Comment (MonicaOlejniczak) [inline packages/core/core/src/requests/AssetGraphRequestRust.js:129]: Did you test the performance difference?

**Comment by mattcompiles** [comment]
Closing in favour of #225

**Pull request closed without merging**